include/libtorrent/sha512.hpp

```diff
@@ -0,0 +1,23 @@
+#ifndef TORRENT_SHA512_HPP_INCLUDED
+#define TORRENT_SHA512_HPP_INCLUDED
+
+#include "libtorrent/config.hpp"
+#include <cstdint>
+
+namespace libtorrent
+{
+
+	struct sha512_ctx
+	{
+		std::uint64_t length, state[8];
+		size_t curlen;
+		std::uint8_t buf[128];
+	};
+
+	TORRENT_EXTRA_EXPORT int SHA512_init(sha512_ctx* context);
```

Comment:
I forget why I did this for SHA1 originally. I think it may have been to mimic libcrypto to make it easy to switch between them. At this point I can't see a good reason to introduce this "intermediate" API, under the ``hasher`` class.

can you think of a reason?